### PR TITLE
LG-12240: Log matched policy OIDs

### PR DESCRIPTION
**Pull request by jc-gsa**
[LG-12240](https://cm-jira.usa.gov/browse/LG-12240)

### CODE REVIEW

**Review by mitchellhenke — APPROVED** [review]

File: spec/controllers/identify_controller_spec.rb [hunk 1]
```diff
@@ -158,11 +158,11 @@
               openssl_errors: 'error 20 at 0 depth lookup: unable to get local issuer certificate',
               ficam_openssl_valid: false,
               ficam_openssl_errors: 'error 20 at 0 depth lookup: unable to get local issuer certificate',
+              matched_policy_oids: ['2.16.840.1.101.2.1.11.9'],
```
Comment (aduth) [inline spec/controllers/identify_controller_spec.rb:161]: I left a question on the ticket in response to your comment about using an array here vs. another structure like a hash and whether we can aggregate on it.

**Review by aduth — APPROVED** [review]
LGTM 👍 

Repeating comment from ticket: 

>I think as long as we're able to search for the OIDs [...] which should be possible with an array. Broadly speaking it'd be nice to see which are the most common generally, but not strictly necessary, and I'm not sure that's it's technically possible at all if multiple matched OIDs are possible.

File: app/controllers/identify_controller.rb [hunk 2]
```diff
@@ -121,6 +121,9 @@ def log_certificate(cert)
     }
 
     attributes.delete(:issuer) if validation_result == 'self-signed cert'
+    if valid
+      attributes[:matched_policy_oids] = cert.matched_policy_oids.map { |item| [item, true] }.to_h
+    end
```
Comment (aduth) [inline app/controllers/identify_controller.rb:126]: There's a lint error here with [`Rails/IndexWith`](https://docs.rubocop.org/rubocop-rails/2.7/cops_rails.html#railsindexwith)

```suggestion
    attributes[:matched_policy_oids] = cert.matched_policy_oids.index_with(true) if valid
```

Comment (jc-gsa) [reply app/controllers/identify_controller.rb]: I've reversed this commit.

Comment (aduth) [reply app/controllers/identify_controller.rb]: Ok, I'm fine with that too. Would the one-liner in the suggestion above work just as well? Dunno if you had a preference.

**Review by mdiarra3 — APPROVED** [review]

File: app/controllers/identify_controller.rb (hunk 2 above)
Comment (aduth) [reply app/controllers/identify_controller.rb]: I was talking about this suggestion: https://github.com/18F/identity-pki/pull/437#discussion_r1523290474

```suggestion
    attributes[:matched_policy_oids] = cert.matched_policy_oids.index_with(true) if valid
```

I expect current commit may cause a lint error.

**Comment by aduth** [comment]
If you rebase this branch against `main` the "check_expiring_certs" job should pass, since this was improved in #443.

File: app/controllers/identify_controller.rb (hunk 2 above)
Comment (aduth) [reply app/controllers/identify_controller.rb]: I guess the lint was fine.

**Review by aduth — APPROVED** [review]

**Pull request merged**